### DISCUSSION TITLE
Should stc necessarily be 1-1 compatible with tsc?

### DISCUSSION THREAD

**Discussion by jakobvase** [dudykr/stc#469] [Ideas]
First of all, thank you for the huge effort you (and others) are putting into this. You're crazy and inspiring :)

I'm opening this because of the following:
1. Typescript is a moving target, and keeping this repository up to date with typescript would require an absolutely enormous amount of effort.
2. I think very many projects would benefit hugely from a stc implementing _most_ of tsc, but being much faster.
3. I think stc can be faster if you choose which parts of tsc to implement, instead of simply going for all of it. (You can ignore some annoying edge cases for example).

So it might make sense to aim for 85 % compatibility or something like that.

**Comment by vacekj** [comment]
This has been on my mind for the past few weeks - my take is `stc` should strive to get adoption from users as soon as possible. Once you get real-world usage, and buy-in from teams, it helps with both funding and further adoption. And for this to happen, stc doesn't need to be at 100% parity with `tsc`, or even close to it. I would argue that most teams using Typescript only use less than 50% of its features. So I would focus on getting the most used (hard to judge, but I think there is some intuition around this) features of `tsc` (or Typescript itself0 done first, and release a Language Server asap. 80% of results for 20% of the work, you know the rule etc. Curious what your thoughts are here @kdy1.

**Reply by kdy1** [reply]
I already had the plan to do it.
See description of https://github.com/dudykr/stc/milestone/1

**Reply by jakobvase** [reply]
Awesome.

**Reply by kdy1** [reply]
Btw, do you have any way to estimate `used` features? Because of `@types/*` packages, lots of features are required.

**Reply by kdy1** [reply]
Those packages use really lots of features of `tsc`

**Reply by jakobvase** [reply]
I would check the version of typescript used in `@types/*` and see how many of them use anything above say 4.3, and then not worry about newer features in typescript at first. My guess is that a lot of them don't use the cutting edge. But then again, my first idea was simply to implement the easy things first, release, and then incrementally add the more complex things - but I have no idea what might be simple or complex to implement :)

**Reply by vacekj** [reply]
Some features are a matter of checking for keywords using regex, some a matter of checking ts version (ruling them out rather) as jakob mentioned above, and some would probs require parsing the source which feels like overkill. 

I would probably look at the top @types/ packages by downloads, and then make sure `stc` works for each of them, starting from the most downloaded ones.

**Reply by kdy1** [reply]
I wrote https://stc.dudy.dev/docs/roadmap

**Reply by kdy1** [reply]
And thank you for the suggestion, I found the problems in my thoughts and concluded that `isolatedModules: true` is required for parallel processing.
